clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -389,6 +389,64 @@ IntType::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// FuncType Definitions
+//===----------------------------------------------------------------------===//
+
+mlir::LogicalResult
+FuncType::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
+                 llvm::ArrayRef<mlir::Type> inputs,
+                 llvm::ArrayRef<mlir::Type> results, bool varArg) {
+  if (results.size() > 1)
+    return emitError() << "functions only supports 0 or 1 results";
+  if (varArg && inputs.empty())
```

Comment:
Nice!

Reply:
Can you please add tests to `invalid.cir` that exercise these verifiers?

Reply:
I've added some tests for this on the following commit since this is a NFC.